### PR TITLE
Allow some N to M siblings transfers

### PR DESCRIPTION
**Pull request by GiudGiud**
@sallustius 
at your own risk this isnt quite ready

### CODE REVIEW

**Comment by GiudGiud** [comment]
@sallustius seems to work for me

File: framework/include/transfers/MultiAppTransfer.h [hunk 1]
```diff
@@ -205,6 +205,13 @@ class MultiAppTransfer : public Transfer
   /// We use the fact that global app indexes are consecutive on a given rank
   unsigned int getLocalSourceAppIndex(unsigned int i_from) const;
 
+  /// Whether the transfer supports siblings transfer
+  virtual void checkSiblingsTransferSupported() const
+  {
+    mooseError("Siblings transfer not supported. You cannot transfer both from a multiapp to "
+               "another multiapp");
+  };
```
Comment (lindsayad) [inline framework/include/transfers/MultiAppTransfer.h:213]:
```suggestion
  }
```

File: framework/src/transfers/MultiAppPostprocessorTransfer.C [hunk 2]
```diff
@@ -70,15 +70,33 @@ MultiAppPostprocessorTransfer::execute()
...
-          if (getToMultiApp()->hasLocalApp(i))
-            getToMultiApp()->appProblemBase(i).setPostprocessorValueByName(_to_pp_name, pp_value);
+          pp_value = from_problem.getPostprocessorValueByName(_from_pp_name);
         }
+
+        // Find the postprocessor value from another process
+        if (getFromMultiApp()->numGlobalApps() == 1)
+          _communicator.min(pp_value);
+        else if (pp_value == std::numeric_limits<Real>::max() && getToMultiApp()->hasLocalApp(i))
+          mooseError("Source and target app parallel distribution must be the same");
```
Comment (lindsayad) [inline framework/src/transfers/MultiAppPostprocessorTransfer.C:85]: can we just do an assertion since you have that check function?

**Review by lindsayad — APPROVED** [review]

File: framework/src/transfers/MultiAppPostprocessorTransfer.C (hunk 2 above)
Comment (GiudGiud) [reply framework/src/transfers/MultiAppPostprocessorTransfer.C]: done

**Comment by GiudGiud** [comment]
Thanks for the review!

**Comment by GiudGiud** [comment]
@sallustius

**Pull request merged**